swap_meet/vendor.py

```diff
@@ -1,2 +1,65 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+    
+    def add(self, item,):
+        self.inventory.append(item)
+        return item 
+    
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item 
+        return False 
+    
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item 
+        return None 
+    
+    def get_by_category(self, category):
+        items_in_category = []
+        for item in self.inventory:
+            if item.get_category() == category:
+                items_in_category.append(item)
+        
+        return items_in_category
+    
+    def get_best_by_category(self, category):
```

Comment:
[suggestion] You could use `get_by_category` in this method. If you did, you could potentially also rely on the `max` function.

Reply:
Thanks so much for this review, Matt! Really appreciate the detailed feedback! Esp looking forward to incorporating list comprehension in future code :)